pkg/syncer/spec/mutators/deployment_test.go

```diff
@@ -490,13 +567,35 @@ func TestMutate(t *testing.T) {
 	} {
 		{
 			t.Run(c.desc, func(t *testing.T) {
+				ctx, cancelFn := context.WithCancel(context.Background())
+				t.Cleanup(cancelFn)
+
 				upstreamURL, err := url.Parse(c.config.Host)
 				require.NoError(t, err)
-				dm := NewDeploymentMutator(upstreamURL)
+
+				scheme := runtime.NewScheme()
+				corev1.AddToScheme(scheme)
+
+				upstreamClient := dynamicfake.NewSimpleDynamicClient(scheme, toObjects(c.upstreamSecrets)...)
+				sharedInformer := dynamicinformer.NewDynamicSharedInformerFactory(upstreamClient, 0)
+				informer := sharedInformer.ForResource(schema.GroupVersionResource{Version: "v1", Resource: "secrets"})
+
+				upstreamSecretIndexer := informer.Informer().GetIndexer()
+				err = upstreamSecretIndexer.AddIndexers(cache.Indexers{
+					ByWorkspaceIndexName: func(obj interface{}) ([]string, error) {
+						return []string{"root:default:testing"}, nil
+					},
+				})
+				require.NoError(t, err)
+
+				sharedInformer.Start(ctx.Done())
+				sharedInformer.WaitForCacheSync(ctx.Done())
```

Comment:
commented before. get rid of all this informer mock code by putting a `List(...) []Secrets` func into the mutator, not a complete lister.

Reply:
done